charts/pulsar/templates/bookkeeper-service.yaml

```diff
@@ -33,7 +33,7 @@ metadata:
 spec:
   ports:
   - name: bookie
-    port: {{ .Values.bookkeeper.ports.bookie }}
```

Comment:
It seems that the keys used in the values should be kept unchanged since it would break backwards compatibility and that's probably also the source of linting failures?

Reply:
@lhotari thanks for the feedback. @addisonj or @sijie Lari raises a great point - is there another way around this? 
Istio also supports setting the `appProtocol` but requires Kubernetes 1.18+
https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection

Reply:
I don't think it breaks any compatibility. Because it can easily upgrade and roll back.

An alternative approach is to add a "portPrefix". By default it is `""`. But users can override it with "tcp-”.

Reply:
It seems that my comment about the keys and backwards compatibility needs clarification. For changing the port names, it shouldn't be necessary to change the content of the values yaml format at all. That's what my original comment was about.

Reply:
I like the portPrefix solution proposed by @sijie. (perhaps it could be called `portNamePrefix` for better clarity)

Reply:
well it seems that it's not "tcp-" for all ports, so perhaps it needs revisiting. in some cases, the prefix is "tls-"

Reply:
Thanks @sijie @lhotari - great idea of some prefixes that by default are blank for backwards compatibility- I'll try to PR that up today